extensions/commands/art/cmd_server.py

```diff
@@ -0,0 +1,195 @@
+                                 f"You can remove it using `conan art:server remove {server_name}`")
+
+
+def assert_existing_server(server_name, servers):
+    server_names = [s["name"] for s in servers]
+    if server_name not in server_names:
+        raise ConanException(f"Server '{server_name}' does not exist.")
+
+
+def add_default_arguments(subparser):
+    subparser.add_argument("name", help="Name of the server")
+    return subparser
+
+
+@conan_command(group="Custom commands")
+def server(conan_api: ConanAPI, parser, *args):
+    """
+    Manages Artifactory server and credentials.
+    """
+
+
+@conan_subcommand()
+def server_add(conan_api: ConanAPI, parser, subparser, *args):
+    """
+    Add Artifactory server and its credentials.
+    """
+    add_default_arguments(subparser)
+    subparser.add_argument("url", help="URL of the artifactory server")
+    subparser.add_argument("--user", help="user name for the repository")
+    subparser.add_argument("--password", help="password for the user name")
+
+    args = parser.parse_args(*args)
+
+    if not args.user:
+        user = input("User: ")
+    else:
+        user = args.user.strip()
+    if not args.password:
+        password = getpass.getpass()
+    else:
+        password = args.password.strip()
+
+    name = args.name.strip()
+    url = args.url.rstrip("/")
+
+    servers = read_servers()
+    assert_new_server(name, servers)
+
+    token = api_request("get", f"{url}/api/security/encryptedPassword", user, password)
+    # TODO: manage error with auth
+
+    new_server = {"name": name,
+                  "url": url,
+                  "user": user,
+                  "password": token}
+    servers.append(new_server)
+    write_servers(servers)
+    ConanOutput().success(f"Server '{name}' ({url}) added successfully")
+
+
+@conan_subcommand()
+def server_remove(conan_api: ConanAPI, parser, subparser, *args):
+    """
+    Remove Artifactory servers.
+    """
+    add_default_arguments(subparser)
+    args = parser.parse_args(*args)
+
+    name = args.name.strip()
+    servers = read_servers()
+    assert_existing_server(name, servers)
+    url = None
+    keep_servers = []
+    for s in servers:
+        if name != s["name"]:
+            keep_servers.append(s)
+        else:
+            url = s["url"]
+    write_servers(keep_servers)
+    ConanOutput().success(f"Server '{name}' ({url}) removed successfully")
+
+
+def output_server_list_text(servers):
+    if servers:
+        for s in servers:
+            ConanOutput().info(f"{s['name']}:")
+            ConanOutput().info(f"  url: {s['url']}")
+            ConanOutput().info(f"  user: {s['user']}")
+            ConanOutput().info(f"  password: *******")
+    else:
+        ConanOutput().info("No servers configured. Use `conan art:server add` command to add one.")
+
+
+def output_server_list_json(servers):
+    result = []
+    for s in servers:
+        s["password"] = "*******"
```

Comment:
I think that if the purpouse of this field is letting users know that the server is authenticated we better use something like: `{.. authenticated: True...}` or something similar? WDYT?